### PR TITLE
docs(readme): improve Cache example

### PR DESCRIPTION
**Pull request by nelson6e65**
Dynamically detect the Store directory and use the latest GitHub actions.

I tested here: https://github.com/elegantech/firebase-odm/runs/6331495061?check_suite_focus=true#step:7:12

This should replace #39 and resolve my question in https://github.com/pnpm/action-setup/issues/8#issuecomment-1119850775

> This also fixes the previous workflow example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,24 +99,41 @@ on:
 
 jobs:
   cache-and-install:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
 
     steps:
-      build:
```
Comment (KSXGitHub) [inline README.md:105]: A valid action should always have a job name.

File: README.md [hunk 2]
```diff
@@ -99,24 +99,41 @@ on:
...
+      - uses: actions/cache@v3
+        name: Setup pnpm cache
+        with:
+          path: ${{ steps.pnpm-cache.outputs.pnpm_cache_dir }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: Install dependencies
+        run: pnpm install
```
Comment (KSXGitHub) [inline README.md:136]: This should be under `build:`

**Review by KSXGitHub — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (nelson6e65) [reply README.md:105]: Hi. I noticed. The job name is `cache-and-install`. In that example was an error.

Comment (nelson6e65) [reply README.md:105]: Because is under `steps`.

File: README.md (hunk 2 above)
Comment (nelson6e65) [reply README.md:136]: @KSXGitHub other workflow examples, like [this](https://github.com/pnpm/action-setup#install-pnpm-and-a-few-npm-packages), are correct. But the cache example was wrong, actually. 😅

Comment (nelson6e65) [reply README.md:136]: The previous example contains this error: image

File: README.md [hunk 3]
```diff
@@ -99,24 +99,41 @@ on:
...
-              ${{ runner.os }}-
-
-        - uses: pnpm/action-setup@v2.1.0
-          with:
-            version: 6.0.2
-            run_install: true
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (KSXGitHub) [inline README.md:108]: Other examples don't have this. Is this necessary for cache? If not, then I think the example should be simplified.

```suggestion
```

**Review by KSXGitHub — CHANGES_REQUESTED** [review]
I have one last question/complaint. Answer/resolve it, and it would be good to merge.

File: README.md (hunk 3 above)
Comment (nelson6e65) [reply README.md:108]: Oh. No, actually. That is specific to keep history for other steps I used in my based project.

I'll remove shortly.

Comment (KSXGitHub) [reply README.md:108]: you just need to use the "commit suggestion" button in the github interface, lol.

File: README.md [hunk 4]
```diff
@@ -99,24 +99,39 @@ on:
 
 jobs:
   cache-and-install:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (KSXGitHub) [inline README.md:102]:
```suggestion
    runs-on: ubuntu-latest
```

File: README.md (hunk 3 above)
Comment (nelson6e65) [reply README.md]: Right! 😅 I just wanted to keep 1 simple commit.

**Review by KSXGitHub — APPROVED** [review]

**Pull request merged**